backend/contributors/danielmadriz/dnd/src/application/schemas.py

```diff
@@ -0,0 +1,68 @@
+"""
+Marshmallow schemas for input validation and output serialization.
+"""
+from marshmallow import Schema, fields, ValidationError, validates_schema, post_load
+from typing import Dict, Any
+
+
+class MonsterRequestSchema(Schema):
+    """Schema for POST /get monster request validation."""
+    monster_index = fields.Str(required=True, validate=lambda x: len(x.strip()) > 0)
+    
+    @validates_schema
+    def validate_monster_index(self, data, **kwargs):
+        """Custom validation for monster_index."""
+        if not data.get('monster_index') or not data['monster_index'].strip():
+            raise ValidationError('monster_index is required and cannot be empty')
```

Comment:
This is kinda duplicated operation since `@validates_schema` is being run after deserialization and field level validation, so on the `monster_index` you're checking if the string is not empty (or just empty spaces), and after that you're running another check that does the same thing (in case of field validation passed). I think cleaner solution would be something like this if you don't want to modify input but enforce correctness instead. 

```suggestion
    monster_index = fields.Str(required=True)
    
    @validates("monster_index")
    def validate_monster_index(self, value):
        if not value.strip():
            raise ValidationError("monster_index is required and cannot be empty")
```

Reply:
Great catch! Fixed in last commit